### PR TITLE
Fix HealthEconomicsDisplay null crashes (Sentry WEB-40/41/42/51)

### PR DESCRIPTION
**Pull request by mikepsinn**
## Summary
- Null-safe guards in HealthEconomicsDisplay for prescriptionAccess / costEffectivenessRating
- Fixes Sentry OPTIMITRON-WEB-40, 51, 41, 42

## Test plan
- [ ] Confirm unit tests pass: `pnpm --filter @optimitron/web exec vitest run src/components/treatment/__tests__/HealthEconomicsDisplay.test.ts`
- [ ] Open a treatment page with health economics where prescriptionAccess fields are null; page should render without crash
- [ ] Open a treatment with unknown/missing costEffectivenessRating; badge should fall back to moderate/outline
- [ ] Spot-check ICER, QALY, and annual cost of care sections still format valid numbers

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented invalid or missing health economics values from appearing in cost, effectiveness, comparator, and prescription-access metrics.
  * Ensured unknown or missing cost-effectiveness ratings use a consistent moderate rating style.

* **Tests**
  * Added coverage for numeric validation and rating-color fallback behavior.

### CODE REVIEW

File: packages/web/src/components/treatment/HealthEconomicsDisplay.tsx [hunk 1]
```diff
@@ -425,34 +451,38 @@ export function HealthEconomicsDisplay({
...
-                    : 'Same cost'}
-                </div>
-              </div>
-              <div className="border-2 border-foreground rounded-md p-3">
-                <div className="text-xs text-muted-foreground font-bold mb-1">QALY Difference</div>
-                <div className="text-lg font-black">
-                  {healthEconomics.vsComparator.qalysGainedDifference > 0 ? '+' : ''}
-                  {healthEconomics.vsComparator.qalysGainedDifference.toFixed(2)} QALYs
+              {isFiniteNumber(healthEconomics.vsComparator.costDifference) && (
+                <div className="border-2 border-foreground rounded-md p-3">
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/web/src/components/treatment/HealthEconomicsDisplay.tsx:460]: Won't fix — this PR only guards null health-economics crashes (Sentry WEB-40/41/42/51). We are not migrating HealthEconomicsDisplay to treaty styling in this change; neobrutalist tokens stay until a dedicated styling PR.

File: packages/web/src/components/treatment/HealthEconomicsDisplay.tsx [hunk 2]
```diff
@@ -64,8 +64,18 @@ const RATING_SWATCHES: Record<CostEffectivenessRating, RatingSwatch> = {
...
+  return typeof value === 'number' && Number.isFinite(value);
+}
+
+export function getCostEffectivenessColors(
+  rating: string | null | undefined,
+): RatingSwatch {
+  if (rating != null && Object.prototype.hasOwnProperty.call(RATING_SWATCHES, rating)) {
+    return RATING_SWATCHES[rating as CostEffectivenessRating];
+  }
+  return RATING_SWATCHES.moderate;
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/web/src/components/treatment/HealthEconomicsDisplay.tsx]: Already fixed in 55397f513 — getCostEffectivenessColors now trims + toLowerCase before RATING_SWATCHES lookup, with title-case coverage in HealthEconomicsDisplay.test.ts.

File: packages/web/src/components/treatment/__tests__/HealthEconomicsDisplay.test.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  });
+
+  it("falls back to moderate for missing or unknown ratings", () => {
+    const fallback = getCostEffectivenessColors("moderate");
+    expect(getCostEffectivenessColors(null).badge).toBe(fallback.badge);
+    expect(getCostEffectivenessColors(undefined).badge).toBe(fallback.badge);
+    expect(getCostEffectivenessColors("unknown").badge).toBe(fallback.badge);
+    expect(getCostEffectivenessColors("").badge).toBe(fallback.badge);
+  });
+});
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/web/src/components/treatment/__tests__/HealthEconomicsDisplay.test.ts:42]: Won't fix — isFiniteNumber (incl. zero) and getCostEffectivenessColors (incl. title-case) already have unit coverage in HealthEconomicsDisplay.test.ts. Full RTL mounts of both variants would add review-driven test surface without catching a distinct remaining bug on this null-guard PR.

File: packages/web/src/components/treatment/HealthEconomicsDisplay.tsx [hunk 4]
```diff
@@ -106,38 +116,46 @@ export function HealthEconomicsDisplay({
...
 
         {/* Additional metrics */}
-        {(healthEconomics.costPerRemission ||
-          healthEconomics.costPerResponder ||
-          healthEconomics.qalysGained ||
-          healthEconomics.icer ||
+        {(isFiniteNumber(healthEconomics.costPerRemission) ||
+          isFiniteNumber(healthEconomics.costPerResponder) ||
+          isFiniteNumber(healthEconomics.qalysGained) ||
+          isFiniteNumber(healthEconomics.icer) ||
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/web/src/components/treatment/HealthEconomicsDisplay.tsx:163]: Won't fix — we are not migrating HealthEconomicsDisplay to treaty styling for this change. Scope stays null-crash guards; brutal tokens remain until a dedicated migration PR.
